lib/_tls_legacy.js

```diff
@@ -426,6 +428,24 @@ CryptoStream.prototype.end = function(chunk, encoding) {
   stream.Duplex.prototype.end.call(this, chunk, encoding);
 };
 
+/*
+ * Wait for both `finish` and `end` events to ensure that all data that
+ * was written on this side was read from the other side.
+ */
+CryptoStream.prototype._destroyWhenReadAndWriteEndsDone = function() {
```

Comment:
I don't think we really need to expose another method on the prototype just to deal with the `finish()` function.